server/game/cards/01_SOR/upgrades/JediLightsaber.ts

```diff
@@ -0,0 +1,41 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { Attack } from '../../../core/attack/Attack';
+import { Card } from '../../../core/card/Card';
+import { UpgradeCard } from '../../../core/card/UpgradeCard';
+import { Trait } from '../../../core/Constants';
+import Player from '../../../core/Player';
+
+export default class JediLightsaber extends UpgradeCard {
+    protected override getImplementationId() {
+        return {
+            id: '8495694166',
+            internalName: 'jedi-lightsaber',
+        };
+    }
+
+    public override canAttach(targetCard: Card, controller: Player = this.controller): boolean {
+        if (targetCard.hasSomeTrait(Trait.Vehicle)) {
+            return false;
+        }
+
+        return super.canAttach(targetCard, controller);
+    }
+
+    public override setupCardAbilities() {
+        this.addGainTriggeredAbilityTargetingAttached({ title: 'Give the defender -2/-2 for this phase',
+            when: { onAttackDeclared: (event, context) => event.attack.attacker === context.source },
+
+            // need to check if the target is a base - if so, don't apply the stat modifier effect
+            immediateEffect: AbilityHelper.immediateEffects.conditional({
+                condition: (context) => context.event.attack.target.isBase(),
+                onTrue: AbilityHelper.immediateEffects.noAction(),
+                onFalse: AbilityHelper.immediateEffects.forThisPhaseCardEffect((context) => ({
+                    effect: AbilityHelper.ongoingEffects.modifyStats({ power: -2, hp: -2 }),
+                    target: (context.event.attack as Attack).target
+                }))
+            }) },
+        (context) => context.source.parentCard?.hasSomeTrait(Trait.Force));
```

Comment:
Good callout, I can refactor so that this is also a part of the properties object. There's a handful of similar updates needed to the upgrade system so I added a TODO and I'll include it in a future PR that hits all of that.